flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClientTest.java

```diff
@@ -250,6 +251,41 @@ public void testCreateFlinkTaskManagerPod() throws Exception {
                 resultTaskManagerPod.getMetadata().getOwnerReferences().get(0).getUid());
     }
 
+    @Test
+    public void testCreateTwoTaskManagerPods() throws Exception {
+        Fabric8FlinkKubeClient localClient =
+                new Fabric8FlinkKubeClient(
+                        flinkConfig,
+                        kubeClient,
+                        Executors.newFixedThreadPool(2, new ExecutorThreadFactory("Testing-IO")));
+        localClient.createJobManagerComponent(this.kubernetesJobManagerSpecification);
+        final KubernetesPod kubernetesPod1 =
+                new KubernetesPod(
+                        new PodBuilder()
+                                .editOrNewMetadata()
+                                .withName("mock-task-manager-pod1")
+                                .endMetadata()
+                                .editOrNewSpec()
+                                .endSpec()
+                                .build());
```

Comment:
This could be factor out into a new method `private KubernetesPod buildTaskManagerPod(String name)` and deduplicate all other similar use cases.